src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx

```diff
@@ -85,12 +86,15 @@ export const ActivityFlowBuilder = () => {
     move(source.index, destination.index);
   };
 
-  const activitiesIds = activities.reduce((acc: Record<string, ActivityFormValues>, activity) => {
-    const id = activity.id || activity.key || '';
-    acc[id] = activity;
+  const activitiesIdsObjects = activities.reduce(
```

Comment:
could you move this functuin into utils?

Reply:
done